streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowTest.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SessionWindowTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowIfEndSmallerThanStart() {
+        new SessionWindow(1, 0);
+    }
+
+    @Test
+    public void testOverlap() {
+        final SessionWindow window = new SessionWindow(50, 100);
```

Comment:
Will change it -- still working to improve my skills to write tests. And you are my "testing hero" right now :)